src/Avalonia.Input/MouseDevice.cs

```diff
@@ -147,15 +119,15 @@ private void ProcessRawEvent(RawPointerEventArgs e)
             if(mouse._disposed)
                 return;
 
-            if (e.Type == RawPointerEventType.NonClientLeftButtonDown) return;
```

Comment:
Instead of ignoring NonClientLeftButtonDown, it's better to handle it the same way as LeaveWindow, and clear pointer over state if necessary + generate leave-events. Added tests for this case.

Reply:
Moved handling of non-client stuff to TopLevel